src/plotter_disk.hpp

```diff
@@ -369,7 +369,9 @@ class DiskPlotter {
         Timer copy;
         do {
             std::error_code ec;
-            if (tmp_2_filename.parent_path() == final_filename.parent_path()) {
+            bool same_filesystem =
+                fs::equivalent(tmp_2_filename.parent_path(), final_filename.parent_path(), ec);
```

Comment:
the name `same_filesystem` make this sound like a broader check than it is. The parent paths may refer to different directories, but still live within the same filesystem.

In order to prefer a rename, when that's possible, I think the most reliable way is to try renaming, and if it fails, fall back to copying.

Reply:
`fs::equivalent` checks whether the paths resolve to the same file system entity, so I think `same_filesystem` is an appropriate name for that variable

Of course we could just try and error, but I think this a more elegant solution.

Reply:
I am sorry, my bad. I misread the documentation of `fs::equivalent` as that it checks whether the two paths resign in the same filesystem.
I updated my pull request to try a rename before falling back to a copy.